src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -784,12 +830,22 @@ public AmazonEC2 connect() throws AmazonClientException {
      * @return {@link AmazonEC2} client
      */
     public static AmazonEC2 connect(AWSCredentialsProvider credentialsProvider, URL endpoint) {
+        if (AmazonEC2Cloud.isTestMode()) {
+            throw new IllegalStateException("This method should not be used in test mode");
+        }
         awsCredentialsProvider = credentialsProvider;
         AmazonEC2 client = new AmazonEC2Client(credentialsProvider, createClientConfiguration(endpoint.getHost()));
         client.setEndpoint(endpoint.toString());
         return client;
     }
 
+    public synchronized static AmazonEC2 mockConnect(AWSCredentialsProvider credentialsProvider, URL endpoint, EC2PrivateKey privateKey) {
```

Comment:
1. Test code should be either package-private or annotated with `@Restricted(NoExternalUse.class)`.
2. This does not seem to be a good way to introduce test code here anyways, though restricting its visibility will make it simpler to refactor.